tests/MyShop.Tests.Integration/Controllers/ProductControllerTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using MyShop.Application.Commands;
+using MyShop.Application.DTO;
+using MyShop.Core.Entities;
+using Shouldly;
+using System.Net;
+using System.Net.Http.Json;
+using Xunit;
+
+namespace MyShop.Tests.Integration.Controllers;
+
+public class ProductControllerTests : ControllerTests, IDisposable
+{
+    private readonly TestDatabase _testDatabase;
+
+    public ProductControllerTests()
+        => _testDatabase = new TestDatabase();
+
+    [Fact]
+    public async Task Post_Product_Should_Return_NoContent_204_Status_Code()
+    {
+        //Assert
+        var category = new Category(Guid.NewGuid(), "Skóra");
+        await _testDatabase.DbContext.Categories.AddAsync(category);
+        await _testDatabase.DbContext.SaveChangesAsync();
+
+        var command = new CreateProduct("Kurtka skórzana", "to dobry produkt", 1000, category.Id);
+
+        //Act
+        var response = await HttpClient.PostAsJsonAsync("product", command);
+
+        //Arrange
+        response.StatusCode.ShouldBe(HttpStatusCode.NoContent);
+    }
+
+    [Fact]
+    public async Task Update_Product_Should_Return_NoContent_204_Status_Code()
+    {
+        //Assert
+        var category = new Category(Guid.NewGuid(), "Skóra");
```

Comment:
ogólnie możesz to dodać razem z produktem bo jak byś dodał to do produktu przed zapisem to automatycznie powinno dodać tez kategorie